### DISCUSSION TITLE
Unique text per test

### DISCUSSION THREAD

**Discussion by lemitrou** [pytest-dev/pytest-xdist#1007] [General]
I'm trying to create objects e.g User class with name, address,email e.t.c with dynamic name, something like user234234 or user2987654,
so I have create Utility class with static method that returns the dynamic text:
`@staticmethod
    def get_random_text(text: str) -> str:
        return text + str(uuid.uuid4())[:8]`
and I have the following fixture:
`@pytest.fixture()
def user_fixture():
    return UserClass(name=Utilities.get_random_text('user'))`

the problem is that when I run tests that call the `user_fixture` in parallel with xdist, in some tests I get the same "random text".
does anybody have any idea why this happens?

**Comment by RonnyPfannschmidt** [comment]
you take too little of the uuid

```
In [28]: for i in range(100000): assert len({uuid.uuid4().hex[:8] for _ in range(20)}) == 20, i
In [29]: for i in range(100000): assert len({uuid.uuid4().hex[:2] for _ in range(20)}) == 20, i
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[29], line 1
----> 1 for i in range(100000): assert len({uuid.uuid4().hex[:2] for _ in range(20)}) == 20, i

AssertionError: 0

In [30]: for i in range(100000): assert len({uuid.uuid4().hex[:3] for _ in range(20)}) == 20, i
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[30], line 1
----> 1 for i in range(100000): assert len({uuid.uuid4().hex[:3] for _ in range(20)}) == 20, i

AssertionError: 5

In [31]: for i in range(100000): assert len({uuid.uuid4().hex[:4] for _ in range(20)}) == 20, i
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[31], line 1
----> 1 for i in range(100000): assert len({uuid.uuid4().hex[:4] for _ in range(20)}) == 20, i

AssertionError: 305
```

please add enough of the random value that its not relatively easy to conflict by accident

**Reply by lemitrou** [reply]
@RonnyPfannschmidt  thanks for the info. It looks that is working very strange though because the total test is not so many, around 100, and the parallel value is 4, so running 4 tests concurrently at the begging phase assign the same value.

**Reply by RonnyPfannschmidt** [reply]
then its possible that the wrong rng is used, and/or the same random source is used in a way that repeats

without more details its hard to tell
